File: vueServer/src/service/general/itDamageService.js
import axios from "axios";

const route = 'general/itdamage';
const myApi = axios.create({
  baseURL: 'http://localhost',
  timeout: 10000,
  withCredentials: true,
  transformRequest: [(data) => JSON.stringify(data.data)],
  headers: {
    'Accept': 'application/json',
    'Content-Type': 'application/json',
  }
});

class itDamageService {
    retrieveList () {
      return myApi.get('/api/' + route);
    }
    retrieve(id) {
      return myApi.get('/api/' + route + '/'+ id);
    }
    update(data) {
      return myApi.put('/api/' + route , data );
    }
    save(data) {
      return myApi.post('/api/' + route + '/save/'+data.rtime, data );
    }
    fileUpload(data) {
      return myApi.put('/api/file/upload/' + data);
    }
    fileDown(data) {
      return myApi.get('/api/file/' + data);
    }
}

export default new itDamageService();

File: src/main/kotlin/com/seohan/erp/general/Service/ItDamageService.java
package com.seohan.erp.general.Service;

import java.util.List;

import com.seohan.erp.general.Domain.ItDamage;

public interface ItDamageService {
	List<ItDamage> itDamage() throws Exception;
	ItDamage save(ItDamage itDamage) throws Exception;
	ItDamage update(ItDamage itDamage) throws Exception; 
}
